From 24d9074c4bf642100c25ddf7f0386b09cd111e09 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 16:14:40 -0400
Subject: [ethernet] be2net: define macro for_all_tx_queues_on_eq()

Message-id: <1410192881-30702-78-git-send-email-ivecera@redhat.com>
Patchwork-id: 92468
O-Subject: [RHEL7.1 PATCH 77/78] be2net: define macro for_all_tx_queues_on_eq()
Bugzilla: 1088575
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>

BZ: #1088575

Upstream commit(s):
commit a4906ea0a5fe6dba89d3809f1e0e97b4d55ef4b7
Author: Sathya Perla <sathya.perla@emulex.com>
Date:   Tue Sep 2 09:56:56 2014 +0530

    be2net: define macro for_all_tx_queues_on_eq()

    Replace the for() loop that traverses all the TX queues on an EQ
    with the macro for_all_tx_queues_on_eq(). With this expalnatory
    name, the one line comment is not required anymore.

    Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/emulex/benet/be.h b/drivers/net/ethernet/emulex/benet/be.h
index d36f8f8..1d9bec8 100644
--- a/drivers/net/ethernet/emulex/benet/be.h
+++ b/drivers/net/ethernet/emulex/benet/be.h
@@ -621,6 +621,10 @@ extern const struct ethtool_ops be_ethtool_ops;
  for (i = eqo->idx, rxo = &adapter->rx_obj[i]; i < adapter->num_rx_qs;\
    i += adapter->num_evt_qs, rxo += adapter->num_evt_qs)
 
+#define for_all_tx_queues_on_eq(adapter, eqo, txo, i)   \
+ for (i = eqo->idx, txo = &adapter->tx_obj[i]; i < adapter->num_tx_qs;\
+  i += adapter->num_evt_qs, txo += adapter->num_evt_qs)
+
 #define is_mcc_eqo(eqo)   (eqo->idx == 0)
 #define mcc_eqo(adapter)  (&adapter->eq_obj[0])
 
diff --git a/drivers/net/ethernet/emulex/benet/be_main.c b/drivers/net/ethernet/emulex/benet/be_main.c
index 68113e4..1528cda 100644
--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -2499,12 +2499,12 @@ int be_poll(struct napi_struct *napi, int budget)
  struct be_adapter *adapter = eqo->adapter;
  int max_work = 0, work, i, num_evts;
  struct be_rx_obj *rxo;
+ struct be_tx_obj *txo;
 
  num_evts = events_get(eqo);
 
- /* Process all TXQs serviced by this EQ */
- for (i = eqo->idx; i < adapter->num_tx_qs; i += adapter->num_evt_qs)
-  be_process_tx(adapter, &adapter->tx_obj[i], i);
+ for_all_tx_queues_on_eq(adapter, eqo, txo, i)
+  be_process_tx(adapter, txo, i);
 
  if (be_lock_napi(eqo)) {
   /* This loop will iterate twice for EQ0 in which
-- 
1.7.1